javascript/javascript3/week3/shoppingCart/main.js

```diff
@@ -0,0 +1,73 @@
+console.log("shopping cart");
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+}
+
+class ShoppingCart {
+  constructor(products) {
+    this.products = products;
+  }
+
+  addProduct(product) {
+    // Implement functionality here
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    if (!product) {
+      return;
+    } else {
+      this.products = this.products.filter(pro => pro.name !== product.name);
+    
+    }
+  }
+
+  searchProduct(productName) {
+    // Implement functionality here
+    if (!productName) {
+      return;
+    } else {
+      this.products = this.products.filter(
+        item => item.name === productName.toLowerCase()
+      );
+    }
+  }
+
+  getTotal() {
+    // Implement functionality here
+    this.totalPrice = this.products.reduce((a, b) => a + b.price, 0);
+  }
+
+  renderProducts() {
+    // Implement functionality here
+    // i'm working on it 
+  }
+
+  getUser() {
```

Comment:
good

Reply:
thank you so much Aleks for the feedback and all the notes that you mentioned.
the cod looks better now (;